oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/CompositeException.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.oak.index.indexer.document;
+
+import org.slf4j.Logger;
+
+public class CompositeException extends Exception {
```

Comment:
What would happen if we also implement "Throwable[] getSuppressed​()" to return the (array of) collectedThrowables? That way it would be more generally useful (I think). I'm not saying this class should be moved to oak-commons right now... also I would still implement the logAllExceptions method, as it looks useful.

Reply:
That method is final, so we can't override it. Adding/getting suppressed exceptions is provided by Throwables class itself. So, keeping this class just for exception type identification and logging method.

Reply:
I see, so instead of overriding getSuppressed() we would have to call addSuppressed(Throwable exception) for each entry in the list. I think it would still make sense do to that, but I don't have a strong opinion.